Update TensorFlow-2.15.1 ml_dtypes dependency to >= 0.3.1 < 0.5.0, so it accepts version 0.4.0

This needs to be applied afer the remove-libclang patch!!

Stamen Miroslavov, 12-09-2024

diff -ruBbw ./tensorflow-2.15.1/tensorflow/tools/pip_package/setup.py ./tensorflow-2.15.1-upgrade/tensorflow/tools/pip_package/setup.py
--- ./tensorflow-2.15.1/tensorflow/tools/pip_package/setup.py	2024-09-13 15:15:40.522498099 +0200
+++ ./tensorflow-2.15.1-upgrade/tensorflow/tools/pip_package/setup.py	2024-09-13 15:31:44.365299214 +0200
@@ -88,7 +88,7 @@
     'gast >=0.2.1,!=0.5.0,!=0.5.1,!=0.5.2',
     'google_pasta >= 0.1.1',
     'h5py >= 2.9.0',
-    'ml_dtypes ~= 0.3.1',
+    'ml_dtypes >=0.3.1, < 0.5.0',
     'numpy >= 1.23.5, < 2.0.0',
     'opt_einsum >= 2.3.2',
     'packaging',
